query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -125,75 +124,99 @@
     if (filter.getChildFilterCount() > 0) {
       String delim = "";
       builder.append("( ");
       for (Filter childFilter : filter.getChildFilterList()) {
         builder.append(delim);
-        builder.append(convertFilter2String(childFilter, paramsBuilder, executionContext));
+        builder.append(convertFilterToString(childFilter, paramsBuilder, executionContext));
         builder.append(" ");
         delim = operator + " ";
       }
       builder.append(")");
     } else {
-      switch (filter.getOperator()) {
-        case LIKE:
-          // The like operation in PQL looks like `regexp_like(lhs, rhs)`
-          Expression rhs =
-              handleValueConversionForLiteralExpression(filter.getLhs(), filter.getRhs());
-          builder.append(operator);
-          builder.append("(");
-          builder.append(
-              convertExpression2String(filter.getLhs(), paramsBuilder, executionContext));
-          builder.append(",");
-          builder.append(convertExpression2String(rhs, paramsBuilder, executionContext));
-          builder.append(")");
-          break;
-        case CONTAINS_KEY:
-          LiteralConstant[] kvp = convertExpressionToMapLiterals(filter.getRhs());
-          builder.append(convertExpressionToMapKeyColumn(filter.getLhs()));
-          builder.append(" = ");
-          builder.append(convertLiteralToString(kvp[MAP_KEY_INDEX], paramsBuilder));
-          break;
-        case CONTAINS_KEYVALUE:
-          kvp = convertExpressionToMapLiterals(filter.getRhs());
-          String keyCol = convertExpressionToMapKeyColumn(filter.getLhs());
-          String valCol = convertExpressionToMapValueColumn(filter.getLhs());
-          builder.append(keyCol);
-          builder.append(" = ");
-          builder.append(convertLiteralToString(kvp[MAP_KEY_INDEX], paramsBuilder));
-          builder.append(" AND ");
-          builder.append(valCol);
-          builder.append(" = ");
-          builder.append(convertLiteralToString(kvp[MAP_VALUE_INDEX], paramsBuilder));
-          builder.append(" AND ");
-          builder.append(MAP_VALUE);
-          builder.append("(");
-          builder.append(keyCol);
-          builder.append(",");
-          builder.append(convertLiteralToString(kvp[MAP_KEY_INDEX], paramsBuilder));
-          builder.append(",");
-          builder.append(valCol);
-          builder.append(") = ");
-          builder.append(convertLiteralToString(kvp[MAP_VALUE_INDEX], paramsBuilder));
-          break;
-        default:
-          rhs = handleValueConversionForLiteralExpression(filter.getLhs(), filter.getRhs());
-          builder.append(
-              convertExpression2String(filter.getLhs(), paramsBuilder, executionContext));
-          builder.append(" ");
-          builder.append(operator);
-          builder.append(" ");
-          builder.append(convertExpression2String(rhs, paramsBuilder, executionContext));
+      if (isAttributeExpressionMapAttribute(filter.getLhs())) {
```

Comment:
The preprocessing step is meant to remove the need for special handling in here.

Reply:
Yes. Will do that.

Reply:
resolved